WalletWasabi/WabiSabi/Client/AmountDecomposer.cs

```diff
@@ -65,12 +65,49 @@ public IEnumerable<Money> Decompose(IEnumerable<Coin> myInputCoins, IEnumerable<
 				}
 			}
 
-			if (remaining >= MinimumAmountPlusFee)
+			if (remaining >= MinimumAmountPlusFee && remainingVsize >= OutputSize)
 			{
-				outputAmounts.Add(remaining - OutputFee);
+				naiveSet.Add(remaining - OutputFee);
+			}
+
+			setCandidates.Add(naiveSet.Count, naiveSet);
+			var before = DateTimeOffset.UtcNow;
+			do
+			{
+				remaining = inputs.Sum();
+				remainingVsize = AvailableVsize;
+				var currSet = new List<Money>();
+				do
+				{
+					var denomPlusFees = denoms.Where(x => x <= remaining && x >= (remaining / 3)).ToList();
+					var denomPlusFee = denomPlusFees.RandomElement();
+					if (denomPlusFee is null || remaining < MinimumAmountPlusFee || remainingVsize < 2 * OutputSize)
+					{
+						break;
+					}
+
+					if (denomPlusFee <= remaining)
+					{
+						currSet.Add(denomPlusFee - OutputFee);
+						remaining -= denomPlusFee;
+						remainingVsize -= OutputSize;
+					}
+				}
+				while (currSet.Count <= naiveSet.Count || currSet.Count <= 3);
+
+				if (currSet.Count <= naiveSet.Count || currSet.Count <= 3)
+				{
+					if (remaining >= MinimumAmountPlusFee && remainingVsize >= OutputSize)
```

Comment:
Again, pointing out the danger here, too in the second condition.

Reply:
We should not consider the decomp valid in that case.

Reply:
https://github.com/zkSNACKs/WalletWasabi/pull/6908/commits/f2cc9bdcf7a378084fd72a5f4bdfc83660d4ca50